File: GennadyCardGame.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    string myCard;
    cin >> myCard;
    vector<string> deck(5);
    for (auto &x:deck)
        cin >> x;
    bool state = false;
    for (int i=0; i<deck.size(); i++) {
        if (myCard[0] == deck[i][0] || myCard[1] == deck[i][1]) {
            state = true;
            break;
        }
        else 
            continue;
    }
    if (state)
        cout << "YES\n";
    else 
        cout << "NO\n";

    return 0;
}

File: Python/GennadyCardGame.py
def main():
    myCard = input()
    deck = [input() for i in range(5)]
    state = False
    for i in range(len(deck)):
        if myCard[0] == deck[i][0] or myCard[1] == deck[i][1]:
            state = True
            break
    if state:
        print("YES")
    else:
        print("NO")


if __name__ == "__main__":
    main()
